app/controllers/spree/admin/payments_controller.rb

```diff
@@ -142,6 +143,17 @@ def can_transition_to_payment
         redirect_to spree.edit_admin_order_customer_url(@order)
       end
 
+      def ensure_sufficient_stock_lines
+        return if !@order.payment? || @order.insufficient_stock_lines.blank?
+
+        out_of_stock_items = @order.insufficient_stock_lines.map do |line_item|
+          line_item.variant.name
+        end.join(", ")
+        flash[:error] = I18n.t("spree.orders.line_item.insufficient_stock",
+          on_hand: "0 #{out_of_stock_items}")
+        redirect_to spree.edit_admin_order_url(@order)
+      end
+
```

Comment:
This week I have been teaching refactoring techniques where "Extract Method" is king.
That out_of_stock_items is a wonderful candidate to be extracted as a method, no params required :ok_hand: